### PR TITLE
Correct fields in Event ( model and api )

### LINKED ISSUE #3816: Correct Event API
- `show_map` -> `is_map_shown`. Change data type to boolean. Default `false`.
- `has_session_speakers` -> `has_sessions_speakers`.
- `ticket_include` -> `is_ticketing_enabled`. (default: true)
- `tax_allow` -> `is_tax_enabled`. 
- `pay_by_paypal` -> `is_pay_by_paypal`
- `pay_by_stripe` -> `is_pay_by_stripe_enabled`
- `pay_by_bank` -> `is_pay_by_bank_enabled`
- `pay_by_cheque` -> `is_pay_by_cheque_enabled`
- `sponsors_enabled` -> `is_sponsors_enabled`
- `email` - remove from API, model and db. Not used.
- `state` - Make all state data lower-case.

Also update db field & model for all of the above without resulting in loss of any data.

### PR DESCRIPTION
**Pull request by LuD1161**
#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [ ] The unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

#### Short description of what this resolves:
Changes some of the fields in Event API and model, changing datatype wherever necessary.

#### Changes proposed in this pull request:
- `show_map` -> `is_map_shown`. Change data type to boolean. Default `false`.
- `has_session_speakers` -> `is_sessions_speakers_enabled`.
- `ticket_include` -> `is_ticketing_enabled`. (default: true)
- `tax_allow` -> `is_tax_enabled`. 
- `pay_by_paypal` -> `can_pay_by_paypal`
- `pay_by_stripe` -> `can_pay_by_stripe`
- `pay_by_bank` -> `can_pay_by_bank`
- `pay_by_cheque` -> `can_pay_by_cheque`
- `pay_onsite` -> `can_pay_onsite`
- `sponsors_enabled` -> `is_sponsors_enabled`
- `email` - remove from API, model and db. Not used.
- `state` - Make all state data lower-case.

Fixes #3816

### CODE REVIEW

File: migrations/versions/3031ff741717_.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+from alembic import op
+import sqlalchemy as sa
+from app.models.event import Event
+from sqlalchemy.orm import sessionmaker
+
+Session = sessionmaker()
+
+
+def upgrade():
+    op.execute('ALTER TABLE events RENAME COLUMN ticket_include TO is_ticketing_enabled')
```
Comment (niranjan94) [inline migrations/versions/3031ff741717_.py]: Why are you executing raw queries for this ? Use alembic's methods to do the column rename please.

File: migrations/versions/3031ff741717_.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+    bind = op.get_bind()
+    session = Session(bind=bind)
+    event = session.query(Event)
+    values = [value.show_map for value in event]
+    counter = 0
+    op.execute('ALTER TABLE events ALTER COLUMN show_map drop default')
+    op.execute('ALTER TABLE events ALTER COLUMN show_map TYPE BOOLEAN USING (show_map::boolean)')
+    op.execute('ALTER TABLE events ALTER COLUMN show_map SET DEFAULT FALSE')
+    for value in event:
```
Comment (niranjan94) [inline migrations/versions/3031ff741717_.py]: Please don't do this ... Use a simple query to do all this for you.

File: migrations/versions/3031ff741717_.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+        value.state = value.state.lower()
+        if values[counter]:
+            value.show_map = True
```
Comment (niranjan94) [inline migrations/versions/3031ff741717_.py]: Use query for this too.

File: migrations/versions/3031ff741717_.py (hunk 1 above)
Comment (LuD1161) [inline migrations/versions/3031ff741717_.py]: Done , working on others .

File: migrations/versions/3031ff741717_.py (hunk 3 above)
Comment (LuD1161) [inline migrations/versions/3031ff741717_.py]: Done :+1:

File: migrations/versions/3031ff741717_.py (hunk 2 above)
Comment (LuD1161) [inline migrations/versions/3031ff741717_.py]: Done :+1:

**Comment by LuD1161** [comment]
@niranjan94  @shubham-padia @SaptakS
Can you review this , please ?

**Comment by hongquan** [comment]
Instead of `is_pay_by_stripe_enabled`, why not `can_pay_by_stripe`?

**Comment by niranjan94** [comment]
@hongquan that sounds much better 😄 ... thanks !!

@LuD1161 do the changes as mentioned above please ...

File: migrations/versions/3031ff741717_.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    op.alter_column('events_version', 'pay_by_stripe', new_column_name='is_pay_by_stripe')
+    op.alter_column('events_version', 'tax_allow', new_column_name='is_tax_enabled')
+    op.alter_column('events_version', 'pay_by_bank', new_column_name='is_pay_by_bank')
+    op.alter_column('events_version', 'pay_by_paypal', new_column_name='is_pay_by_paypal')
+    op.alter_column('events_version', 'has_session_speakers', new_column_name='has_sessions_speakers')
+    op.alter_column('events_version', 'sponsors_enabled', new_column_name='is_sponsors_enabled')
+    op.alter_column('events_version', 'pay_by_cheque', new_column_name='is_pay_by_cheque')
+    op.drop_column('events_version', 'email')
+    op.execute('ALTER TABLE events ALTER COLUMN show_map drop default')
+    op.execute('ALTER TABLE events ALTER COLUMN show_map TYPE BOOLEAN USING (show_map::boolean)')
```
Comment (niranjan94) [inline migrations/versions/3031ff741717_.py]: Also, set default value as TRUE for this

File: migrations/versions/3031ff741717_.py [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+    op.alter_column('events_version', 'is_tax_enabled', new_column_name='tax_allow')
+    op.alter_column('events_version', 'is_pay_by_bank', new_column_name='pay_by_bank')
+    op.alter_column('events_version', 'is_pay_by_paypal', new_column_name='pay_by_paypal')
+    op.alter_column('events_version', 'has_sessions_speakers', new_column_name='has_session_speakers')
+    op.alter_column('events_version', 'is_sponsors_enabled', new_column_name='sponsors_enabled')
+    op.alter_column('events_version', 'is_pay_by_cheque', new_column_name='pay_by_cheque')
+    op.add_column('events_version', sa.Column('email', sa.String(), nullable=True))
+
+    op.execute('ALTER TABLE events ALTER COLUMN is_map_shown drop default')
+    op.execute('ALTER TABLE events ALTER COLUMN is_map_shown TYPE INTEGER USING (is_map_shown::integer)')
```
Comment (niranjan94) [inline migrations/versions/3031ff741717_.py]: Set the default value as `1` after this.

**Comment by LuD1161** [comment]
@hongquan @niranjan94 
Changed .
`pay_by_cheque` => `can_pay_by_cheque`
`pay_by_stripe` => `can_pay_by_stripe`
`pay_by_paypal` => `can_pay_by_paypal`
`pay_by_bank` => `can_pay_by_bank`

File: migrations/versions/3031ff741717_.py (hunk 5 above)
Comment (LuD1161) [inline migrations/versions/3031ff741717_.py]: Done :+1:

File: migrations/versions/3031ff741717_.py (hunk 4 above)
Comment (LuD1161) [inline migrations/versions/3031ff741717_.py]: Done :+1:

File: app/api/events.py [hunk 6]
```diff
@@ -63,21 +62,21 @@ class Meta:
...
     payment_currency = fields.Str()
     paypal_email = fields.Str()
-    tax_allow = fields.Bool(default=False)
-    pay_by_paypal = fields.Bool(default=False)
-    pay_by_stripe = fields.Bool(default=False)
-    pay_by_cheque = fields.Bool(default=False)
-    pay_by_bank = fields.Bool(default=False)
+    is_tax_enabled = fields.Bool(default=False)
+    is_pay_by_paypal = fields.Bool(default=False)
+    is_pay_by_stripe = fields.Bool(default=False)
```
Comment (niranjan94) [inline app/api/events.py]: Still not changed. Please pay more attention to things.

Comment (LuD1161) [inline app/api/events.py]: @niranjan94
Done .

**Comment by LuD1161** [comment]
@niranjan94 
please review.

File: migrations/versions/76c949b1235f_the_events_model_and_api_migrations.py [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+    op.alter_column('events_version', 'pay_by_paypal', new_column_name='can_pay_by_paypal')
+    op.alter_column('events_version', 'pay_by_cheque', new_column_name='can_pay_by_cheque')
+    op.alter_column('events_version', 'pay_onsite', new_column_name='can_pay_onsite')
+    op.alter_column('events_version', 'tax_allow', new_column_name='is_tax_enabled')
+    op.alter_column('events_version', 'ticket_include', new_column_name='is_ticketing_enabled')
+    op.alter_column('events_version', 'has_session_speakers', new_column_name='has_sessions_speakers')
+    op.alter_column('events_version', 'sponsors_enabled', new_column_name='is_sponsors_enabled')
+    op.drop_column('events_version', 'email')
+    op.execute('ALTER TABLE events ALTER COLUMN show_map drop default')
+    op.execute('ALTER TABLE events ALTER COLUMN show_map TYPE BOOLEAN USING (show_map::boolean)')
```
Comment (niranjan94) [inline migrations/versions/76c949b1235f_the_events_model_and_api_migrations.py]: Do the same for the `events_version` table too

Comment (LuD1161) [inline migrations/versions/76c949b1235f_the_events_model_and_api_migrations.py]: @niranjan94 
Done :+1:

**Comment by LuD1161** [comment]
@niranjan94
Can you review it now ?

**Comment by LuD1161** [comment]
@niranjan94
Sorry, I didn't thought about it earlier.
Have done it now.

On Jun 19, 2017 15:31, "Niranjan Rajendran" <notifications@github.com>
wrote:

*@niranjan94* requested changes on this pull request.

These need to be updated in API docs too. Please always do it.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/fossasia/open-event-orga-server/pull/3820#pullrequestreview-44800745>,
or mute the thread

**Comment by LuD1161** [comment]
@niranjan94
Updated that.

**Comment by niranjan94** [comment]
Since 

- `ticket_include` -> `is_ticketing_enabled`
- `has_session_speakers` -> `has_sessions_speakers`.
- `sponsors_enabled` -> `is_sponsors_enabled`

Since all of the above do similar function, lets also rename `has_session_speakers` to `is_sessions_speakers_enabled` for uniformity. 

@LuD1161 sorry for the multiple changes.

**Comment by LuD1161** [comment]
@niranjan94 
Did all changes. Final list of changes - 

- `show_map` -> `is_map_shown`. Change data type to boolean. Default `false`.
- `has_session_speakers` -> `is_sessions_speakers_enabled`.
- `ticket_include` -> `is_ticketing_enabled`. (default: true)
- `tax_allow` -> `is_tax_enabled`. 
- `pay_by_paypal` -> `can_pay_by_paypal`
- `pay_by_stripe` -> `can_pay_by_stripe`
- `pay_by_bank` -> `can_pay_by_bank`
- `pay_by_cheque` -> `can_pay_by_cheque`
- `pay_onsite` -> `can_pay_onsite`
- `sponsors_enabled` -> `is_sponsors_enabled`
- `email` - remove from API, model and db. Not used.
- `state` - Make all state data lower-case.

Have made changes in the docs too .

**Comment by LuD1161** [comment]
@niranjan94 
The travis build error is due to **tests**, cause those also need to be refactored.

The particular test file , I am pointing towards is **/tests/unittests/object_mother.py**.

**Check this** 
`grep -r 'has_session_speakers' .`

image

But that's not the only place to look at, there are many other files using the same variable , so if a model is refactored we need to check it's usage in all the files.

Recently there have been some other refactoring , for example  [PR#3814](https://github.com/fossasia/open-event-orga-server/pull/3814)  **( Refactor event api/model )**

Here the **logo** has been changed to **logo_url** 
image

Similarly for **background_url** , it has been changed to **original_image_url** , but there are references to it all over.
**Check this**
image

Similarly for the **user model refactoring** in [PR#3812](https://github.com/fossasia/open-event-orga-server/pull/3812) 
```
firstname => first_name
avatar => avatar_url
facebook => facebook_url
```
And many other similar changes.

image

For this PR I'm going to refactor the test and it should work. In both of these code changes , there wasn't any link in the **tests** , so it must have passed the Travis build.
But this issue needs to be solved.
What do you suggest should be done , 
- Make a new issue for the endpoints refactored upto now and 
- For new refactoring issues , it should be taken care of in the PR itself .

Or do you suggest something different ?

**Comment by niranjan94** [comment]
Doing a direct grep on the project dir is gonna give you loads of files. because we still have the old code base inside the `vintage` directory. We have old migration files. 

All you have to do is, only fix the files that matter. Which in this case are the test cases. Why are you concerned about stuff in vintage, migrations and compiled pyc files ?

**Comment by LuD1161** [comment]
@niranjan94 
That's not what I'm concerned about but what about these **/app/helpers/data.py** , **/app/helpers/data_getter.py**  ?

**Comment by niranjan94** [comment]
@LuD1161 are you following other PRs and conversation on gitter ? I even remember a message from @shubham-padia asking everyone not to use methods inside any of the helpers at `app/helpers` and instead of they need to use any method there, the method needs to be moved to `app/api/helper`. 

No .. what the above means is, none of the methods inside `app/helpers/data.py` , `app/helpers/data_getter.py` should be used anywhere. So need not be refactored. At the end of nextgen refactor, we'll delete those files anyway

**Comment by LuD1161** [comment]
@niranjan94 
ok , I must have missed it

File: docs/api/api_blueprint.apib [hunk 8]
```diff
@@ -564,7 +562,49 @@ Get a list of events.
 
 
 ### Create Event [POST]
-Create a new event using a `name`, `starts-at`, `ends-at`, `timzone` and an optional content body.
+Create a new event using a `name`, `starts-at`, `ends-at`, `timezone` and an optional content body.
+
++ name (string) - Name of the event **(required)**
```
Comment (niranjan94) [inline docs/api/api_blueprint.apib]: Why did you remove the table structure and updated description added in https://github.com/fossasia/open-event-orga-server/pull/3832 ?

File: docs/api/api_blueprint.apib [hunk 9]
```diff
@@ -824,6 +862,47 @@ Update a single event.
 All other fields are optional. Add attributes you want to modify.
 
 + `id` (integer) - ID of the record to update **(required)**
++ name (string)
```
Comment (niranjan94) [inline docs/api/api_blueprint.apib]: Why did you remove the table structure and updated description added in https://github.com/fossasia/open-event-orga-server/pull/3832 ?

File: tests/unittests/object_mother.py [hunk 10]
```diff
@@ -22,7 +22,7 @@ def get_event():
                      location_name='India',
                      topic='Travel & Outdoor',
                      sub_topic='Climbing',
-                     has_session_speakers=True,
```
Comment (niranjan94) [inline tests/unittests/object_mother.py]: How does `has_session_speakers` become `is_ticketing_enabled`

File: docs/api/api_blueprint.apib (hunk 9 above)
Comment (LuD1161) [inline docs/api/api_blueprint.apib]: @niranjan94 
sorry , that was a mistake.

File: tests/unittests/object_mother.py (hunk 10 above)
Comment (LuD1161) [inline tests/unittests/object_mother.py]: @niranjan94 
sorry , that was a mistake.

Comment (LuD1161) [inline tests/unittests/object_mother.py]: @niranjan94 
Can you review it ?

File: docs/api/api_blueprint.apib [hunk 11]
```diff
@@ -581,8 +579,7 @@ Create a new event using a `name`, `starts-at`, `ends-at`, `timzone` and an opti
                   "identifier": "1234",
                   "name": "New Event",
                   "starts-at": "2017-06-01",
-                  "ends-at": "2017-06-05",
-                  "email": "example@example.com"
+                  "ends-at": "2017-06-05"
```
Comment (niranjan94) [inline docs/api/api_blueprint.apib]: Datetime format is wrong

File: docs/api/api_blueprint.apib [hunk 12]
```diff
@@ -581,8 +579,7 @@ Create a new event using a `name`, `starts-at`, `ends-at`, `timzone` and an opti
                   "identifier": "1234",
                   "name": "New Event",
                   "starts-at": "2017-06-01",
```
Comment (niranjan94) [inline docs/api/api_blueprint.apib]: Datetime format is wrong

Comment (LuD1161) [inline docs/api/api_blueprint.apib]: Done :+1:

File: docs/api/api_blueprint.apib (hunk 11 above)
Comment (LuD1161) [inline docs/api/api_blueprint.apib]: Done :+1:

**Comment by niranjan94** [comment]
This looks good to me ✅ . But lets have a second review on all PRs. 

@fossasia/open-event-server-devs please review this so that we can merge

**Comment by niranjan94** [comment]
@LuD1161 please update the API docs for this.
